### DISCUSSION TITLE
Huge performance hit when accessed via Microsoft Remote Desktop (RDP)

### DISCUSSION THREAD

**Discussion by jassonmc** [kimai/kimai#4415] [General] [answered]
### Describe the issue

I am using Kimai v2.2.1 and wanted to access it via a Microsoft Remote Desktop Server.
There is a massive performance hit, as soon as I open any entry, that opens up a menu in front and blurs the background, like creating a new time entry.
I used two different Kimai instances, one installed via Docker and another one installed via a shared hosting. Both are v2.2.1 and end up with the same issue.
I am pretty sure that it's the blur effect, which is causing server performance to its knees, it's really that extreme and it happens when creating a new time entry, a new customer, a new project or when modifying an entry.

Firefox browser has the most extreme hit, server becomes unusable. Edge browser or Brave also generate a huge hit, but not as much, still renders the Remote Desktop Server kinda unusable.

Is there a way to deactivate that background blur effect globally in Kimai, so I could make some further testing?

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

2.2.1

### How do you run Kimai?

Docker

### Which PHP version are you using?

8.2

**Comment by kevinpapst** [comment]
Is there are reason you post everything twice? Doesn't make reading easier...
This is no bug in Kimai as well, merely a discussion => moving.

Start by tuning your browser settings. Not that I am an expert in that area, but I am pretty sure that every browser has different performance and graphic related settings.

https://superuser.com/questions/387980/scrolling-browser-through-rdp-is-jerky-other-applications-behave-fine-why

**Reply by jassonmc** [reply]
Oh sorry, that got strangely saved...
Anyway, removed the double content.

Thanks for your suggestion in the link. Although the infos there are more then 10 years old, I did try to adjust RDP quality settings in various ways, but to no avail, no change no matter if I used ultra low or ultra high quality settings. 
I deactivated the hardware acceleration in Firefox and Brave, but that didn't make any difference either.
I did try then on two newer Remote Desktop Server 2022 and interestingly, on one server the performance hit was remarkably lower, but not on the other. I could also see the difference in the "quality" of the background blur effect and it was a lot less washed out and I could read the text in the background, whereas on the Remote Desktop Server with the high cpu impact, background could not be read.

While I'll surely try to further dig into this, as to why one server works ok and the technically identical server not, it is clear to me, that the background blur effect is the culprit for some reason.

Is there any simple "work around" you could think of to at least temporarily disable that effect in order to understand, if that's really the problem?

**Reply by kevinpapst** [reply]
> Is there any simple "work around" you could think of 

Nope, sorry. 
This is standard Bootstrap modal behavior. 
But you could test if the problem is also existing on https://getbootstrap.com/docs/5.3/components/modal/

**Reply by jassonmc** [reply]
I figured, that the property "backdrop-filter: blur (4px)" is stored 4 times in each of those 2 files:
public/bundles/tabler/tabler-rtl.css
public/bundles/tabler/tabler.css

I did replace 4px with 0px in those files, which basically will disable the blur.
Unfortunately, even after reloading Kimai cache, Firefox insepctor still shows me 4px in tabler.css...

Is there anything else I could do to tell Kimai to "reinitialize" the contents of tabler.css?

I know, that this won't be upgrade safe, but for now I am simply interested in a quick fix for the time being :)

**Reply by jassonmc** [reply]
Never mind, did completely clean out Firefox cache and now it works, no more blur :)
Is there a way to save my hack in a config somehow, so it could become update safe?

Thanks a lot already for guiding me to the right place.

**Reply by kevinpapst** [reply]
Try the CustomCSS bundle and add a rule that uses `!important` to overwrite the blur. Maybe that works.

**Reply by jassonmc** [reply]
You're the man, that worked flawlessly :)
Installed plugin CustomCSS bundle and added the following CSS:

/* START [general] deactivate-modal-blur */
.modal-blur { -webkit-backdrop-filter: blur(0px) !important;backdrop-filter: blur(0px) !important; }
/* END [general] deactivate-modal-blur */

Works like a charm.

Thanks again for an amazing piece of software and your great help, really appreciate it.

**Reply by kevinpapst** [reply, marked as answer]
Thanks for sharing 👍 
I added it as example here https://www.kimai.org/documentation/plugin-custom-css.html

<img width="701" alt="Bildschirmfoto 2023-11-09 um 15 46 02" src="https://github.com/kimai/kimai/assets/533162/66bfde72-f168-4350-b948-24e4e0b4f963">

**Comment by jassonmc** [comment]
Thanks for the link to bootstrap.
I just analyzed via Firefox inspector the blur and found the relevant property "backdrop-filter: blur (4px);"
See the printscreen attached.
As soon as I disable that property, the blur vanishes and the issue with insane high cpu load as well.

Unfortunately, I'm no programmer, so if you could guide me to where in the Kimai installation I would need to make the necessary adjustment to disable the blur would be a huge help.

Thanks already for pointing to the right direction :)

![Kimai-Blur-Effect](https://github.com/kimai/kimai/assets/31354299/f4c6b5c6-198d-417b-bfd2-d575d1a2cf74)
